source/custom/nirfmxinstr_service.custom.cpp

```diff
@@ -62,4 +62,50 @@ ::grpc::Status NiRFmxInstrService::GetNIRFSASessionArray(::grpc::ServerContext*
     return ::grpc::Status(::grpc::NOT_FOUND, ex.what());
   }
 }
+
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+// BuildPortString is custom implemented to account for a bug in the buffer size returned from the ivi-dance call to BuildPortString2.
+// We adjust the resulting size from the ivi-dance call to account for the channel_number input.
+// See AB#1835966 for details about the RFmx Instr Bug.
+::grpc::Status NiRFmxInstrService::BuildPortString(::grpc::ServerContext* context, const BuildPortStringRequest* request, BuildPortStringResponse* response)
+{
+  if (context->IsCancelled()) {
+    return ::grpc::Status::CANCELLED;
+  }
+  try {
+    char* selector_string = (char*)request->selector_string().c_str();
+    char* port_name = (char*)request->port_name().c_str();
+    char* device_name = (char*)request->device_name().c_str();
+    int32 channel_number = request->channel_number();
+
+    while (true) {
+      auto status = library_->BuildPortString(selector_string, port_name, device_name, channel_number, 0, nullptr);
+      if (status < 0) {
+        response->set_status(status);
+        return ::grpc::Status::OK;
+      }
+      int32 selector_string_out_length = status + std::to_string(channel_number).length();  // AB#1835966: RFmx Instr BuildPortString2 Bug.
+
+      std::string selector_string_out;
+      if (selector_string_out_length > 0) {
+        selector_string_out.resize(selector_string_out_length - 1);
+      }
+      status = library_->BuildPortString(selector_string, port_name, device_name, channel_number, selector_string_out_length, (char*)selector_string_out.data());
+      if (status == kErrorReadBufferTooSmall || status == kWarningCAPIStringTruncatedToFitBuffer || status > static_cast<decltype(status)>(selector_string_out_length)) {
+        // buffer is now too small, try again
+        continue;
+      }
+      response->set_status(status);
+      if (status == 0) {
```

Comment:
Generated code uses status_ok which checks status>=0 not status==0

Reply:
Ah, good catch. Updated to status>=0.